### PR TITLE
Add `annotate_allele_info` function to `utils.annotations`

### PR DESCRIPTION
**Pull request by jkgoodrich**

### CODE REVIEW

**Review by mike-w-wilson — COMMENTED** [review]
A couple comments on the docs and a couple questions: This maintains the `nonsplit_alleles=ht.old_alleles`. Is this used downstream? We dropped this in v3.1 release since it could contain alleles belonging to unreleased samples. Also looks like the ac_group_filter is in this PR too so you may want to merge main in?

File: gnomad/utils/annotations.py [hunk 1]
```diff
@@ -800,6 +800,55 @@ def add_variant_type(alt_alleles: hl.expr.ArrayExpression) -> hl.expr.StructExpr
...
+        This function requires that the input `ht` is unsplit, and a split `ht` is
+        returned.
+
+    'allele_info' is a struct with the following information:
+        - variant_type - Variant type (snv, indel, multi-snv, multi-indel, or mixed).
+        - n_alt_alleles - Total number of alternate alleles observed at variant locus.
+        - has_star - True if the variant contains a star allele.
+        - allele_type - Allele type (snv, insertion, deletion, or mixed).
+        - was_mixed - True if the variant was mixed (i.e. contained both SNVs and indels).
+        - nonsplit_alleles - Array of alleles before splitting.
```
[the remark this answers is not in the document]
Comment (jkgoodrich) [reply gnomad/utils/annotations.py]: I think this one will, but I changed the note for annotate_freq so it looks better

**Comment by jkgoodrich** [comment]
> A couple comments on the docs and a couple questions: This maintains the `nonsplit_alleles=ht.old_alleles`. Is this used downstream? We dropped this in v3.1 release since it could contain alleles belonging to unreleased samples. Also looks like the ac_group_filter is in this PR too so you may want to merge main in?

Yup, I would like to have it on this annotation HT, but removed from the actual release HT

**Comment by jkgoodrich** [comment]
Checks should pass after [this PR](https://github.com/broadinstitute/gnomad_methods/pull/540) goes in and I merge with main

**Review by mike-w-wilson — APPROVED** [review]
LGTM

**Pull request merged**